### PR TITLE
[MM-39640] Migrate from gorp to sqlx in store/sqlstore/role_store.go

### PR DESCRIPTION
**Pull request by shadowshot-x**
#### Summary

Migrating from gorp to sqlx in role_store.go.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/18878

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/role_store.go
Comment (isacikgoz) [inline store/sqlstore/role_store.go:181]: This applies to similar occurrences;
```suggestion
	dbRoles := []Role{}
```

**Review by isacikgoz — COMMENTED** [review]
Thanks for the contribution @shadowshot-x , I think we better initialize the slices upfront.

File: store/sqlstore/role_store.go
Comment (shadowshot-x) [reply store/sqlstore/role_store.go:181]: Right @isacikgoz. Did not notice that. Correcting it.

File: store/sqlstore/role_store.go
Comment (agnivade) [inline store/sqlstore/role_store.go:228]: This is okay to keep as-is.

File: store/sqlstore/role_store.go
Comment (agnivade) [inline store/sqlstore/role_store.go:247]: `var role Role` is fine.

**Review by agnivade — COMMENTED** [review]
Looking great @shadowshot-x. Just some minor adjustments.

File: store/sqlstore/role_store.go
Comment (shadowshot-x) [reply store/sqlstore/role_store.go:228]: Sure @agnivade. Doing this.

**Review by agnivade — APPROVED** [review]
Nice work @shadowshot-x !

**Comment by agnivade** [comment]
Thanks @shadowshot-x - this looks good from my side. I was waiting for Ibrahim's approval, but he has been buried in other stuff. We can merge this. Sorry for the delay!

**Pull request merged**
